src/frontend/flask_server.py

```diff
@@ -241,7 +241,7 @@ def _submit_transaction(transaction_data):
     try:
         resp.raise_for_status() # Raise on HTTP Status code 4XX or 5XX
     except requests.exceptions.HTTPError as err:
-        raise UserWarning(resp.json().get('msg', ''))
+        raise UserWarning(resp.text)
```

Comment:
This issue seems to be present on line 271 too. Do you mind copying this change over there?

Reply:
Made the changes